src/OAuth1/Provider/Atlassian.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * SocialConnect project
+ * @author: Andreas Heigl https://github.com/heiglandreas <andreas@heigl.org>
+ */
+
+namespace SocialConnect\OAuth1\Provider;
+
+use SocialConnect\Common\Http\Client\Client;
+use SocialConnect\Common\Http\Client\ClientInterface;
+use SocialConnect\OAuth1\Signature\MethodRSASHA1;
+use SocialConnect\Provider\AccessTokenInterface;
+use SocialConnect\Provider\Consumer;
+use SocialConnect\Provider\Exception\InvalidResponse;
+use SocialConnect\OAuth1\AbstractProvider;
+use SocialConnect\Common\Entity\User;
+use SocialConnect\Common\Hydrator\ObjectMap;
+use SocialConnect\Provider\Session\SessionInterface;
+
+class Atlassian extends AbstractProvider
+{
+    /**
+     * @var string The Base-URI of the Atlassian instance
+     */
+    private $baseUri;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBaseUri()
+    {
+        return $this->baseUri;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAuthorizeUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/authorize';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequestTokenUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/request-token';
+    }
+
+    /**
+     * @return string
+     */
+    public function getRequestTokenAccessUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/access-token';
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'atlassian';
+    }
+
+    /**
+     * Atlassian constructor.
+     *
+     * Required configuration parameters are:
+     *
+     * * base_uri The base URI of your self-hosted atlassian product
+     * * private_key The path to the private key file used for signing.
```

Comment:
Is this correct? A quick search doesn't show any use of `$parameters['private_key']` in the code.